libs/s25main/figures/nofWorkman.cpp

```diff
@@ -55,7 +55,7 @@ void nofWorkman::HandleStateWaiting2()
 {
     current_ev = nullptr;
     // Ware erzeugen... (noch nicht "richtig"!, sondern nur viruell erstmal)
-    if(!(ware = ProduceWare()).has_value())
+    if(!(ware = ProduceWare()).has_value() || workplace->is_emptyCycle)
```

Comment:
As explained above: I think it makes more sense to put the emptyCycle member into the miner class and handle it in `ProduceWare`